java/core/src/test/java/sleeper/core/range/canonicaliser/RangeCanonicaliserByteArrayTypeTest.java

```diff
@@ -0,0 +1,124 @@
+import sleeper.core.schema.Schema;
+import sleeper.core.schema.type.ByteArrayType;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+public class RangeCanonicaliserByteArrayTypeTest {
+    private Field field;
+    private RangeFactory rangeFactory;
+
+    @Before
+    public void setup() {
+        field = new Field("key", new ByteArrayType());
+        Schema schema = Schema.builder().rowKeyFields(field).build();
+        rangeFactory = new RangeFactory(schema);
+    }
+
+    @Test
+    public void shouldCanonicaliseRangeMinInclusiveMaxExclusive() {
+        //Given
+        Range range = rangeFactory.createRange(field, new byte[]{1}, new byte[]{10});
+
+        //When
+        Range canonicalisedRange = RangeCanonicaliser.canonicaliseRange(range);
+
+        //Then
+        assertThat(range.isInCanonicalForm()).isTrue();
+        assertThat(canonicalisedRange).isEqualTo(range);
+    }
+
+    @Test
+    public void shouldCanonicaliseRangeMinInclusiveMaxInclusive() {
+        //Given
+        Range range = rangeFactory.createRange(field, new byte[]{1}, true, new byte[]{10}, true);
+
+        //When
+        Range canonicalisedRange = RangeCanonicaliser.canonicaliseRange(range);
+
+        //Then
+        assertThat(range.isInCanonicalForm()).isFalse();
+        assertThat(canonicalisedRange).isEqualTo(rangeFactory.createRange(field, new byte[]{1}, nextByteArrayValue(new byte[]{10})));
+    }
+
+    @Test
+    public void shouldCanonicaliseRangeMinExclusiveMaxInclusive() {
+        //Given
+        Range range = rangeFactory.createRange(field, new byte[]{1}, false, new byte[]{10}, true);
+
+        //When
+        Range canonicalisedRange = RangeCanonicaliser.canonicaliseRange(range);
+
+        //Then
+        assertThat(range.isInCanonicalForm()).isFalse();
+        assertThat(canonicalisedRange).isEqualTo(rangeFactory.createRange(field, nextByteArrayValue(new byte[]{1}), nextByteArrayValue(new byte[]{10})));
+    }
+
+    @Test
+    public void shouldCanonicaliseRangeMinExclusiveMaxExclusive() {
+        //Given
+        Range range = rangeFactory.createRange(field, new byte[]{1}, false, new byte[]{10}, false);
+
+        //When
+        Range canonicalisedRange = RangeCanonicaliser.canonicaliseRange(range);
+
+        //Then
+        assertThat(range.isInCanonicalForm()).isFalse();
+        assertThat(canonicalisedRange).isEqualTo(rangeFactory.createRange(field, nextByteArrayValue(new byte[]{1}), new byte[]{10}));
+    }
+
+    @Test
+    public void shouldHandleRangeWithMaxNullExclusive() {
+        //Given
+        Range range = rangeFactory.createRange(field, new byte[]{1}, true, null, false);
+
+        //When
+        Range canonicalisedRange = RangeCanonicaliser.canonicaliseRange(range);
+
+        //Then
+        assertThat(range.isInCanonicalForm()).isTrue();
+        assertThat(canonicalisedRange).isEqualTo(range);
+    }
+
+    @Test
+    public void shouldHandleRangeWithMaxNullInclusive() {
```

Comment:
It would be helpful if the test name could illustrate why the range is already canonicalised in this case. Maybe something like shouldSetRangeToMaxExclusiveWhenSpecifiedAsNullInclusive?